htmd/parameterization/dihedral.py

```diff
@@ -242,39 +253,72 @@ def _vectorToParams(self, vector):
 
         return parameters
 
-    def _optimizeWithRandomSearch(self, vector):
+    def _optimizeWithIterativeScheme(self, vector, target_energies):
```

Comment:
This is getting a spaghetti code. i.e `self._optimizeWithRandomSearch` should not be coerced to do local optimizations and other stuff not related wih NRS.

Just replicate the creation of local optimizers (could be factored out in the future) and use them directly. Random sampling is also just a few lines of code.

Reply:
I made the Iterative scheme now optimize directly instead of setting the num_searches to 1. I refactored out the two optimizers which also avoids the previous re-initialization of optimizers.